### PR TITLE
adding discount fields

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: app/users/controllers/AdminViewSet.py [hunk 1]
```diff
@@ -37,6 +37,8 @@ def get_permissions(self):
             self.permission_classes = [IsAuthenticated, IsSuperUser]
         elif self.action == "list":
```
Comment (MohamedMegdich1) [inline app/users/controllers/AdminViewSet.py:38]: why adding this when in else i have the isAuthentification ermission

File: app/subscription/services/SubscriptionPlanService.py [hunk 2]
```diff
@@ -76,6 +100,32 @@ def update_subscription_plan(cls, instance, data, partial=False):
...
+                discount_percentage is None
+                or start_date is None
+                or end_date is None
+                or discount_message is None
+            ):
+                raise serializers.ValidationError(
+                    "Discount percentage, start date, end date and discount meesage are required \
+                    when discount is true."
+                )
+            if float(discount_percentage) < 0 or float(discount_percentage) > 100:
```
Comment (MohamedMegdich1) [inline app/subscription/services/SubscriptionPlanService.py:121]: the validation is set it in the modal

File: app/subscription/services/SubscriptionPlanService.py [hunk 3]
```diff
@@ -76,6 +100,32 @@ def update_subscription_plan(cls, instance, data, partial=False):
...
+
+        discount = data.get("discount", instance.discount)
+        discount_percentage = data.get("discount_percentage", instance.discount_percentage)
+        start_date = data.get("start_date", instance.start_date)
+        end_date = data.get("end_date", instance.end_date)
+        discount_message = data.get("discount_message", instance.discount_message)
+
+        if discount:
+            if (
+                discount_percentage is None
```
Comment (MohamedMegdich1) [inline app/subscription/services/SubscriptionPlanService.py:112]: refacto this

**Review by MohamedMegdich1 — CHANGES_REQUESTED** [review]

File: app/users/signals.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+@receiver(post_delete, sender=Admin)
+@receiver(post_delete, sender=Architect)
+@receiver(post_delete, sender=Client)
+@receiver(post_delete, sender=Supplier)
+def delete_associated_user(sender, instance, **kwargs):
+    """
+    Deletes the associated ArchimatchUser when the user type instance is deleted.
+    """
+    if instance.user:
+        instance.user.delete()
```
Comment (MohamedMegdich1) [inline app/users/signals.py:23]: don't delete the user add a field is_deleted and test on it

File: app/subscription/models/SubscriptionPlan.py [hunk 5]
```diff
@@ -29,6 +45,27 @@ def save(self, *args, **kwargs):
...
+        Custom validation method to ensure valid discount fields.
+        """
+        if self.discount:
+            if (
+                self.discount_percentage is None
+                or self.start_date is None
+                or self.end_date is None
+                or not self.discount_message
+            ):
+                raise serializers.ValidationError(
```
Comment (MohamedMegdich1) [inline app/subscription/models/SubscriptionPlan.py:59]: change the message

**Review by MohamedMegdich1 — CHANGES_REQUESTED** [review]

File: app/users/signals.py (hunk 4 above)
Comment (ahmedhamilastage) [reply app/users/signals.py:23]: why keep the user and set is_deleted , 
if so every presence of user model should be modified to check if is_deleted , and when readding user also
why not just delete it ?

Comment (MohamedMegdich1) [reply app/users/signals.py:23]: we should keep it for data integrity and auditing ,accidental deletion

**Review by MohamedMegdich1 — APPROVED** [review]

**Pull request merged**